pkg/controllers/metrics/pod/controller.go

```diff
@@ -132,13 +141,19 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			Labels:   labels,
 		},
 	})
-	c.recordPodStartupMetric(pod)
+	c.recordPodStartupMetric(pod, labels)
 	return reconcile.Result{}, nil
 }
 
-func (c *Controller) recordPodStartupMetric(pod *corev1.Pod) {
+func (c *Controller) recordPodStartupMetric(pod *corev1.Pod, labels prometheus.Labels) {
 	key := client.ObjectKeyFromObject(pod).String()
 	if pod.Status.Phase == phasePending {
+		cond, ok := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
```

Comment:
If the pod has the scheduled condition, will it also be pending? Don't you need to do something like track that the pod _was_ pending and then know that it has now been marked as scheduled